Frontend/DinoTracker/app/controllers/login.ts

```diff
@@ -2,27 +2,34 @@ import Controller from "@ember/controller";
 import { tracked } from "@glimmer/tracking";
 import { action } from '@ember/object';
 import { inject as service } from "@ember/service";
-import mutation from "dino-tracker/graphql/mutations/login.graphql";
+import ApolloRepository from "dino-tracker/services/apollo-repository";
 
 export default class LoginController extends Controller {
-  @service apollo: any;
   @service router: any;
+  @service apolloRepository!: ApolloRepository;
 
   @tracked username = '';
   @tracked password = '';
   @tracked isDisplayingErrorMessage = false;
 
   @action
-  authenticate() {
+  async authenticate()
+  {
     let variables = { username: this.username, password: this.password };
-    this.get('apollo').mutate({ mutation, variables }, "login")
-      .then((response: any) => {
-        if (response == true)
-          this.router.transitionTo("staff");
-        else
-          this.isDisplayingErrorMessage = true;
-      })
-      .catch(() => this.isDisplayingErrorMessage = true);
+    let response: boolean;
+
+    try
+    {
+      response = await this.get('apolloRepository').login(variables);

```

Comment:
Nice!

Reply:
FYI, you can just access services directly: `this.apolloRepository.login(...)` rather than using get. Don't worry about it here though